src/groupederror.jl

```diff
@@ -1,15 +1,14 @@
 ##### List of functions to analyze data
 
 function extend_axis(df::AbstractDataFrame, xlabel, ylabel, xaxis, val)
-    aux = DataFrame()
-    aux[xlabel] = xaxis
+    aux = DataFrame(Any[xaxis],[xlabel])
     extended = join(aux, df, on = xlabel, kind = :left)
     sort!(extended, cols = [xlabel])
     return convert(Array, extended[ylabel], val)
 end
 
 function extend_axis(xsmall, ysmall, xaxis, val)
-    df = DataFrame(x = xsmall, y = ysmall)
+    df = DataFrame(Any[xsmall,ysmall], [:x,:y])
```

Comment:
Thanks for taking care of the bug. Still, I'll probably keep this somewhat uglier version that works no matter what until NullableArrays stabilizes, and then I'll go over the code again to see if it can be made more elegant.